code/code/misc/multiclass.cc

```diff
@@ -150,42 +150,18 @@ int TBeing::getClassNum(classIndT arg) const
 }
 
 
-classIndT TBeing::getClassIndNum(const char *arg, exactTypeT exact) const
-{
-  int which = getClassNum(arg, exact);
-  classIndT res = MIN_CLASS_IND;
-
-  for(classIndT i=MIN_CLASS_IND;i<MAX_CLASSES;i++){
-    if(classInfo[i].class_num==which){
-      res=i;
-      break;
-    }
-  }
-      
-  if(res==MIN_CLASS_IND) {
-    vlogf(LOG_BUG, "unknown class result");
-  }
+classIndT TBeing::getClassIndNum(const char *arg, exactTypeT exact) const {
+  auto which = static_cast<unsigned short>(getClassNum(arg, exact));
```

Comment:
`getClassNum` returns `classInfo[i].class_num`. `classInfo` is an array of `struct class_info`, which declares `class_num` as `int`.

```c++
struct class_info
{
  bool enabled;
  classIndT class_lev_num;
  int class_num;
  sstring name;
  discNumT base_disc;
  discNumT sec_disc;
  double prac_tweak;
  double hp_per_level;
  sstring abbr;
};
```

`classInfo` is initialized in [`constants.cc`](https://github.com/sneezymud/sneezymud/blob/ea68bb8c5b6449d4e4813dede958f7f084c4e38b/code/code/misc/constants.cc#L983):

```c++
const struct class_info classInfo[MAX_CLASSES] =
{
  {true, MAGE_LEVEL_IND, CLASS_MAGE, "mage", 
   DISC_MAGE, DISC_LORE, 0.43, 7.5 * 35.0 / 50.0, "M"},

  {true, CLERIC_LEVEL_IND, CLASS_CLERIC, "cleric",
   DISC_CLERIC, DISC_THEOLOGY, 0.47, 8.0 * 35.0 / 50.0, "C"},

  {true, WARRIOR_LEVEL_IND, CLASS_WARRIOR, "warrior",
   DISC_WARRIOR, DISC_NONE, 0.47, 8.5, "W"},

  {true, THIEF_LEVEL_IND, CLASS_THIEF, "thief",
   DISC_THIEF, DISC_NONE, 0.41, 8.0 * 35.0 / 50.0, "T"},

  {true, SHAMAN_LEVEL_IND, CLASS_SHAMAN, "shaman",
   DISC_SHAMAN, DISC_NONE, 0.39, 7.5 * 35.0 / 50.0, "S"},

  {true, DEIKHAN_LEVEL_IND, CLASS_DEIKHAN, "deikhan",
   DISC_DEIKHAN, DISC_THEOLOGY, 0.44, 7.5, "D"},

  {true, MONK_LEVEL_IND, CLASS_MONK, "monk",
   DISC_MONK, DISC_NONE, 0.44, 7.5 * 35.0 / 50.0, "K"},

  {false, RANGER_LEVEL_IND, CLASS_RANGER, "ranger",
   DISC_RANGER, DISC_NONE, 0.46, 7.0 * 35.0 / 50.0, "R"},

  {false, COMMONER_LEVEL_IND, CLASS_COMMONER, "commoner",
   DISC_ADVENTURING, DISC_NONE, 0.40, 5, "O"}
};
```
The `CLASS_X` constants used to initialize the `class_num` property of the struct are type `unsigned short`:

![image](https://user-images.githubusercontent.com/7330404/168185560-8d836c1a-4779-4310-ba4a-6811994fe520.png)

Which as you suspected is also the type of the `Class` property in the `charFile` class:

```c++
class charFile {
  public:
   byte sex;
   byte level[MAX_SAVED_CLASSES];
   ubyte race;
   byte screen;
   attack_mode_t attack_type;
   float weight;
   unsigned short height;
   byte  doneBasic[MAX_SAVED_CLASSES];
   unsigned short Class;
```

So that's my really long way of saying, "yes you are correct". Changing the type of `class_info::class_num` to `unsigned short` is probably the logical change here, I'm assuming? Is that actually worth the effort?

Reply:
Nah. I'd rather get the playerfiles to DB first.